istio-control/istio-discovery/templates/telemetryv2.yaml

```diff
@@ -1,5 +1,234 @@
 {{- if and .Values.telemetry.enabled .Values.telemetry.v2.enabled }}
-{{ .Files.Get "metadata-exchange-v2.yaml" }}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: metadata-exchange
+  {{- if .Values.global.configRootNamespace }}
+  namespace: {{ .Values.global.configRootNamespace }}
+  {{- else }}
+  namespace: {{ .Release.Namespace }}
+  {{- end }}
+spec:
+  configPatches:
+    - applyTo: HTTP_FILTER
+      match:
+        context: ANY # inbound, outbound, and gateway
+        proxy:
+          proxyVersion: {{ .Values.telemetry.v2.proxyMatch.proxyVersion }}
```

Comment:
@mandarjog  How are we going to deal with upgrade? Another set of envoyfilter for post 1.4 proxy version?

Reply:
the envoyfilter templates are compiled in together with istioctl so they are versioned under the hood already? @ostromart WDYT?

Reply:
This is not how it will work in practice. The envoyfilter is global, but during an upgrade you will have multiple versions of a proxy running which is where it breaks

Reply:
but if the proxyMatch part we can still select proxies whose versions are compatible with the envoyFilters?

Reply:
Yeah, you will need to make both envoyfilter 1.4 and 1.5 installed to not break user.

Reply:
We have 2 choices.
1. Encode the version in the name, so those Configs can coexist.
2. Leave it as is, but the next version of filters configs will have extra config patches of the previous version. After the api stabilizes I don’t expect we will need this so much.

If we go with option 1, this pr needs to be updated.

Reply:
I am ok with first option